src/vcpkg/base/system.process.cpp

```diff
@@ -895,18 +896,21 @@ namespace vcpkg
         }
 
         Debug::print(proc_id, ": popen(", actual_cmd_line, ")\n");
+
+        const char* cmd_str = actual_cmd_line.c_str();
         // Flush stdout before launching external process
         fflush(stdout);
 
-        const auto pipe = popen(actual_cmd_line.c_str(), "r");
+        const auto pipe = popen(cmd_str, "r");
         if (pipe == nullptr)
         {
             return format_system_error_message("popen", errno);
         }
 
-        char buf[1024];
+        char buf[1024] = {0};
```

Comment:
```suggestion
        char buf[1024];
```

I don't think zeroing this is useful given that we immediately overwrite it with `fgets`.